### PR TITLE
spline #925 add unique constraint validation

### PR DESCRIPTION
**Pull request by wajda**
Add unique constraint validation on ExecutionPlan operation IDs, attribute IDs and expression IDs

### CODE REVIEW

File: commons/src/main/scala/za/co/absa/spline/common/validation/ValidationUtils.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  @throws[ConstraintViolationException]
+  def validate(c: Constraint): Unit = {
+    validate(c.isValid, c.message)
+  }
+
+  @throws[ConstraintViolationException]
+  def validate(b: Boolean, msg: => String): Unit = {
+    if (!b) throw new ConstraintViolationException(msg, null) // NOSONAR
+  }
```
Comment (cerveada) [inline commons/src/main/scala/za/co/absa/spline/common/validation/ValidationUtils.scala:31]: - I don't understand why is this separated method. Could both validate methods be just merged into one?
- the name doesn't describe what the method does. Maybe `throwIfNotValid` would make more sense

Comment (wajda) [reply commons/src/main/scala/za/co/absa/spline/common/validation/ValidationUtils.scala:31]: It's a utility object that aims for the samilar purpose as JSR-380, but in a simplistic implementation without annotations and external dependencies. The method is called `validate()` as the standard `Validator.validate()`, and it's overloaded to support simple checks that don't necessarily require a dedicated constraint class. E.g.
```scala
ValidationUtils.validate(age > 0, s"Invalid age: $age")
```

Comment (cerveada) [reply commons/src/main/scala/za/co/absa/spline/common/validation/ValidationUtils.scala:31]: ok, used like that it makes more sense

**Review by cerveada — APPROVED** [review]
LGTM

**Pull request merged**